altos-rust/altos-core/src/sched/mod.rs

```diff
@@ -112,110 +135,227 @@ pub fn start_scheduler() {
-    start_scheduler();
-    assert!(test::current_task().is_some());
-    assert_eq!(handle_1.tid(), Ok(test::current_task().unwrap().tid()));
-
-    switch_context();
-    assert!(test::current_task().is_some());
-    assert_eq!(handle_2.tid(), Ok(test::current_task().unwrap().tid()));
-
-    switch_context();
-    assert!(test::current_task().is_some());
-    assert_eq!(handle_1.tid(), Ok(test::current_task().unwrap().tid()));
-  }
-
-  #[test]
-  fn test_scheduler_higher_first() {
-    let _g = test::set_up();
-    assert!(test::current_task().is_none());
-    let _handle_1 = test::create_and_schedule_test_task(512, Priority::Normal, "test task 1");
-    let handle_2 = test::create_and_schedule_test_task(512, Priority::Critical, "test task 2");
-    start_scheduler();
-    assert!(test::current_task().is_some());
-    assert_eq!(handle_2.tid(), Ok(test::current_task().unwrap().tid()));
-
-    switch_context();
-    assert!(test::current_task().is_some());
-    assert_eq!(handle_2.tid(), Ok(test::current_task().unwrap().tid()));
-
-    switch_context();
-    assert!(test::current_task().is_some());
-    assert_eq!(handle_2.tid(), Ok(test::current_task().unwrap().tid()));
-  }
-
-  #[test]
-  fn test_scheduler_lower_if_higher_blocked() {
-    let _g = test::set_up();
-    assert!(test::current_task().is_none());
-    let handle_1 = test::create_and_schedule_test_task(512, Priority::Normal, "test task 1");
-    let handle_2 = test::create_and_schedule_test_task(512, Priority::Critical, "test task 2");
-    start_scheduler();
-    assert!(test::current_task().is_some());
-    assert_eq!(handle_2.tid(), Ok(test::current_task().unwrap().tid()));
-    test::current_task().unwrap().state = State::Blocked;
-    test::current_task().unwrap().delay_type = Delay::Timeout;
-
-    switch_context();
-    assert!(test::current_task().is_some());
-    assert_eq!(handle_1.tid(), Ok(test::current_task().unwrap().tid()));
-
-    switch_context();
-    assert!(test::current_task().is_some());
-    assert_eq!(handle_1.tid(), Ok(test::current_task().unwrap().tid()));
-  }
-
-  #[test]
-  fn test_scheduler_doesnt_schedule_destroyed_tasks() {
-    let _g = test::set_up();
-    assert!(test::current_task().is_none());
-    let mut handle_1 = test::create_and_schedule_test_task(512, Priority::Normal, "test task 1");
-    let mut handle_2 = test::create_and_schedule_test_task(512, Priority::Normal, "test task 2");
-    let handle_3 = test::create_and_schedule_test_task(512, Priority::Normal, "test task 3");
-    start_scheduler();
-    assert!(test::current_task().is_some());
-    assert_eq!(handle_1.tid(), Ok(test::current_task().unwrap().tid()));
-
-    switch_context();
-    assert!(test::current_task().is_some());
-    assert_eq!(handle_2.tid(), Ok(test::current_task().unwrap().tid()));
-    handle_1.destroy();
+    use super::*;
+    use test;
+
+    #[test]
+    //Board does not have any tasks scheduled on boot
```

Comment:
I'm not so sure these comments really add anything considering the names of our tests